clients/src/test/java/org/apache/kafka/common/config/AbstractConfigTest.java

```diff
@@ -276,52 +277,46 @@ public RestrictedClassLoader() {
         ClassLoader restrictedClassLoader = new RestrictedClassLoader();
         ClassLoader defaultClassLoader = AbstractConfig.class.getClassLoader();
 
-        // Test default classloading where all classes are visible to thread context classloader
-        Thread.currentThread().setContextClassLoader(defaultClassLoader);
-        ClassTestConfig testConfig = new ClassTestConfig();
-        testConfig.checkInstances(ClassTestConfig.DEFAULT_CLASS, ClassTestConfig.DEFAULT_CLASS);
-
-        // Test default classloading where default classes are not visible to thread context classloader
-        // Static classloading is used for default classes, so instance creation should succeed.
-        Thread.currentThread().setContextClassLoader(restrictedClassLoader);
-        testConfig = new ClassTestConfig();
-        testConfig.checkInstances(ClassTestConfig.DEFAULT_CLASS, ClassTestConfig.DEFAULT_CLASS);
-
-        // Test class overrides with names or classes where all classes are visible to thread context classloader
-        Thread.currentThread().setContextClassLoader(defaultClassLoader);
-        ClassTestConfig.testOverrides();
-
-        // Test class overrides with names or classes where all classes are visible to Kafka classloader, context classloader is null
-        Thread.currentThread().setContextClassLoader(null);
-        ClassTestConfig.testOverrides();
-
-        // Test class overrides where some classes are not visible to thread context classloader
-        Thread.currentThread().setContextClassLoader(restrictedClassLoader);
-        // Properties specified as classes should succeed
-        testConfig = new ClassTestConfig(ClassTestConfig.RESTRICTED_CLASS, Collections.singletonList(ClassTestConfig.RESTRICTED_CLASS));
-        testConfig.checkInstances(ClassTestConfig.RESTRICTED_CLASS, ClassTestConfig.RESTRICTED_CLASS);
-        testConfig = new ClassTestConfig(ClassTestConfig.RESTRICTED_CLASS, Arrays.asList(ClassTestConfig.VISIBLE_CLASS, ClassTestConfig.RESTRICTED_CLASS));
-        testConfig.checkInstances(ClassTestConfig.RESTRICTED_CLASS, ClassTestConfig.VISIBLE_CLASS, ClassTestConfig.RESTRICTED_CLASS);
-        // Properties specified as classNames should fail to load classes
+        ClassLoader origin = Thread.currentThread().getContextClassLoader();
         try {
-            new ClassTestConfig(ClassTestConfig.RESTRICTED_CLASS.getName(), null);
-            fail("Config created with class property that cannot be loaded");
```

Comment:
It's good refactor to use `assertThrow`, but the failed message is missed. Please add it in `assertThrow`. Thanks.